libs/csm/rest/csm_rest_query_deployment.py

```diff
@@ -161,30 +174,264 @@ def get_node_topology(self, node_id=None, auth_header=None):
+        """
+        Function to verify deployment version and time
+        """
+        err_msg = ""
+        time_list = []
+        result = True
+        self.log.info("Verify deployment version")
+        response_version = get_response.json()["topology"]["version"]
+        resp = self.master.execute_cmd(
+                cmd=cmds.K8S_POD_INTERACTIVE_CMD.format(self.pod_list[0], cmds.CMD_CORTX_VERSION),
+                read_lines=True)
+        resp = resp.decode() if isinstance(resp, bytes) else resp
+        deploy_version = [res.strip().split()[1] for res in resp]
+        deploy_version = ''.join(deploy_version).strip('\"')
+        self.log.info("Deploy_version %s ", deploy_version)
+        if response_version == deploy_version:
+            self.log.info("Deployment Version match found")
+        else:
+            err_msg = "Deployment Version mismatch found. "
+            self.log.error(err_msg)
+            result = False
+        self.log.info("Verify deployment time of nodes")
+        for dicts in get_response.json()["topology"]["nodes"]:
+            if 'deployment_time' in dicts.keys():
+                time_list.append(dicts['deployment_time'])
+        self.log.info("List of deployment times: %s", time_list)
+        for times in time_list:
+            if deploy_start_time <= times <= deploy_end_time:
+                self.log.info("Deployment time verification successful")
+            else:
+                err_msg = err_msg + "Deployment time verification failed"
+                self.log.error(err_msg)
                 result = False
+        return result, err_msg
+
+    def verify_node_details(self, get_response: dict):
+        """
+        Verify node details
+        """
+        err_msg = ""
+        result = True
+        self.log.info("Check if all key parameters are present for node")
+        list_params = ['id', 'version', 'services', 'type', 'storage_set',
+                'deployment_time', 'hostname']
+        for dicts in get_response.json()["topology"]["nodes"]:
+            for param in list_params:
+                self.log.info("Node param: %s", param)
+                if param in dicts.keys() and dicts[param] != "":
+                    self.log.info("Node param verification successful")
+                else:
+                    err_msg = "Node param verification failed"
+                    self.log.error(err_msg)
+                    result = False
+        return result, err_msg
+
+    def verify_storage_details(self, get_response: dict):
+        """
+        Verify storage details present for data node
+        """
+        solution_yaml = self.get_solution_yaml()
+        storage1 = solution_yaml["solution"]["storage_sets"][0]["storage"]
+        for dicts in get_response.json()["topology"]["nodes"]:
+            self.log.info("dict is: %s ", dicts)
+            if 'storage' in dicts.keys() and dicts["storage"][0]["name"] == "cvg-01":
```

Comment:
There can be more than 2 cvg. 
we want to compare the response of csm json and solution.yaml.
cvg-01 should not be hardcoded